GitUI/CommandsDialogs/RevisionDiff.cs

```diff
@@ -32,13 +32,15 @@ public partial class RevisionDiff : GitModuleControl
         private GitItemStatus _oldDiffItem;
         private IRevisionDiffController _revisionDiffController;
         private readonly IFullPathResolver _fullPathResolver;
+        private readonly IFindFilePredicateProvider _findFilePredicateProvider;
 
         public RevisionDiff()
         {
             InitializeComponent();
             Translate();
             this.HotkeysEnabled = true;
             _fullPathResolver = new FullPathResolver(() => Module.WorkingDir);
+            _findFilePredicateProvider = new FindFilePredicateProvider(() => Module.WorkingDir);
```

Comment:
I think you had it "more" correct before - `Module.WorkingDir` represents the state, so was rightfully passed as an input parameter.
Behaviours are injected via the constructor, so it should be `new FindFilePredicateProvider(() => Module);`.

Right now, there is no consistent way of injecting or passing `GitModule` in because it is a combination of state and behavior, and they needs to be separated. This is a big and complex piece of work which is brewing on a back burner...

Reply:
Ok, i returned it back.